tools/niminst/install.tmpl

```diff
@@ -3,7 +3,20 @@
 #  result = "#! /bin/sh\n# Generated by niminst\n"
 #  var proj = c.name.toLower
 
-set -e
+BASE_DIR=$(pwd)
+NIM_DIR=$(cd -P -- "$(dirname -- "$(command -v -- "$0")")" && pwd -P)
```

Comment:
What does -P do? what does -- do? What does $(dirname -- $(command -v -- $0) && pwd -P)  do? Please keep in mind that the more I forget about Unix the happier I get.

Please comment this mess.